### PR TITLE
Registration through REST API with additional profile fields

### LINKED ISSUE #489: Allow additional profile fields to be included into the signup process
I agree with this ticket, we should mimic what we are making available from front end from the BP REST API.

See: https://buddypress.trac.wordpress.org/ticket/8976

### PR DESCRIPTION
**Pull request by renatonascalves**
Ticket: https://buddypress.trac.wordpress.org/ticket/8976

fixes #489

### CODE REVIEW

File: includes/bp-members/classes/class-bp-rest-signup-endpoint.php [hunk 1]
```diff
@@ -402,10 +396,99 @@ public function create_item( $request ) {
 		// Hash and store the password.
 		$meta['password'] = wp_hash_password( $password );
 
-		$user_name = $request->get_param( 'user_name' );
```
Comment (imath) [inline includes/bp-members/classes/class-bp-rest-signup-endpoint.php:405]: I'm not sure how deprecated fields are handled within the REST API, but I would probably keep this part to check if a `user_name` was sent within the request to eventually log a deprecated notice.

Then a bit later I'd check if the new way of getting signup fields generated a not empty $meta['field_1']. If $meta['field_1'] is empty, I'd probably try to set it using the `$user_name`.

**Review by imath — APPROVED** [review]
Hi @renatonascalves 

Thanks for your great work about it. I think we should go the way you improved the signup endpoint as it's consistent with what's happening on the web version.

I believe we simply need to make sure deprecated usage is traced for the developer so that he can update his code. That's why I added my inline comment as I'm unsure how the REST API is managing deprecated fields.

Once you committed, you can also write a developer note about the changes on bpdevel.wordpress.com to early inform about this coming change in BuddyPress 14.0

File: includes/bp-members/classes/class-bp-rest-signup-endpoint.php (hunk 1 above)
Comment (renatonascalves) [reply includes/bp-members/classes/class-bp-rest-signup-endpoint.php:405]: If the data is not sent, or it it sent empty, without `user_name`, the REST API request will fail. Meaning that there is no possibility for the data to fallback to `user_name`.

Any required field will fail a request, just like in the web version. So the user will either to fix their request to add it and/or to remove the `user_name` field.

Comment (renatonascalves) [reply includes/bp-members/classes/class-bp-rest-signup-endpoint.php:405]: The only way out for consumers here is to migrate to `signup_field_data`.

Comment (renatonascalves) [reply includes/bp-members/classes/class-bp-rest-signup-endpoint.php:405]: If you look at how the code is setup, having to account for this field as a fallback would be more trouble than it is worth, IMO. And we would lose the main goal here: have parity with the web version.

**Comment by renatonascalves** [comment]
> Once you committed, you can also write a developer note about the changes on bpdevel.wordpress.com to early inform about this coming change in BuddyPress 14.0

I'll do this. 💪🏾

File: includes/bp-members/classes/class-bp-rest-signup-endpoint.php (hunk 1 above)
Comment (renatonascalves) [reply includes/bp-members/classes/class-bp-rest-signup-endpoint.php:405]: Done: https://bpdevel.wordpress.com/2024/05/07/signup-fields-via-the-rest-api/

**Pull request merged**